context.go

```diff
@@ -16,14 +16,41 @@ import (
 	"github.com/sdboyer/gps"
 )
 
+type BuildContext interface {
```

Comment:
I like the principle here, but IMO `DepContext` is preferable to `BuildContext`, as we're not really _building_ anything here.